realscale.py

```diff
@@ -99,12 +189,31 @@ def effect(self):
 
         p_length = self.getUnittouu(str(distance((p1_x, p1_y),(p2_x, p2_y))) + self.getDocumentUnit())
         
+        # Find Drawing Scale
+        if self.options.choosescale == 'scales':
+            drawnscale = int(self.options.scales)
+        elif self.options.choosescale == 'metric':
+            drawnscale = int(self.options.metric)
+        elif self.options.choosescale == 'imperial':
+            drawnscale = int(self.options.imperial)
+        elif self.options.choosescale == 'manual':
+            drawnscale = int(self.options.drawnscale)
+        
         # calculate scaling center
         center = self.calc_scale_center(p1_x, p1_y, p2_x, p2_y)
 
         #calculate scaling factor
         target_length = self.getUnittouu(str(self.options.length) + self.options.unit)
-        factor = target_length / p_length
+        factor = (target_length / p_length) / drawnscale
+        #inkex.debug("%s, %s  %s" % (target_length, p_length, factor))
+        
+        # Add scale rule
+        if self.options.showscale == "true":
+            dist = int(self.options.unitlength)
+            # !! this isn't working right if drawnscale != 1 (1:1)
+            ruler_length = self.getUnittouu(str(dist*2) + self.options.unit) / drawnscale #* drawnscale * factor
+            ruler_pos = (p1_x + (p2_x - p1_x)/2, (p1_y + (p2_y - p1_y)/2) - self.getUnittouu('4 mm'))
+            self.create_ruler( self.document.getroot(), ruler_length/drawnscale, ruler_pos, dist, drawnscale)
```

Comment:
Does it make sense to divide the rulerlength value the user chooses by the drawing scale twice? I'll need to play around with this I guess...